### ISSUE TITLE
Use SQLite for persistence

### ISSUE THREAD

**Issue by cglotr** [hooligram/hooligram-client#57]
Subtasks:
- [x] #64 design local device database
- [ ] #73 implement user & message tables
- [ ] [#78]() design migration strategy (tentative only, to be discussed and decided on)
- [ ] [server#33](https://github.com/hooligram/hooligram-server/issues/33) fetch migration version from backend 

Things to consider:
* Migration. Interesting pattern: https://github.com/andpor/react-native-sqlite-storage/issues/157
* Version update checking & enforcement #69
* Reconnection

**Comment by cglotr** [comment]
This might be related to this issue.

I think we should detach messaging from redux store. Reasoning:
1. Messaging is not related to the state of the app. Having 0 or any messaging will not affect the state of the app. Almost the same argument with removing forms state from the app
2. JavaScript object is not supposed to replace DB. Messaging size will grow `O(n)` and DB is meant for this use case.
3. Redux requires not to mutate the object which makes it more inefficient for DB like data persistence load.
4. Pagination problem. Let's say the message is 1k long (possible for long running message group).
5. We should make the store as small as possible.
6. Redux to DB sync problem. Instead of figuring out complex ways of syncing, let's just drop redux store.

Basically, the app should talk to the DB directly (like making a HTTP request for the data). Since the DB is local, speed is never going to be an issue
